.github/workflows/build-nightly.yml

```diff
@@ -37,19 +42,16 @@ jobs:
             latest=auto
           tags: |
             type=raw,value=nightly
-            type=schedule,pattern=nightly-{{date 'YYYY-MM-DD' tz='Etc/UTC'}}
+            type=schedule,pattern=${{ steps.version_vars.outputs.mastodon_nightly_version }}
           labels: |
             org.opencontainers.image.description=Nightly build image used for testing purposes
 
-      - name: Generate version suffix
-        id: version_vars
-        run: |
-          echo mastodon_version_suffix=+nightly-$(date +'%Y%m%d') >> $GITHUB_OUTPUT
-
       - uses: docker/build-push-action@v4
         with:
           context: .
-          build-args: MASTODON_VERSION_SUFFIX=${{ steps.version_vars.outputs.mastodon_version_suffix }}
+          # The version suffix needs the plus character here so that it reads as
+          # v4.1.2+nightly-2022-03-05 instead of v4.1.2nightly-2022-03-05
+          build-args: MASTODON_VERSION_SUFFIX=+${{ steps.version_vars.outputs.mastodon_nightly_version }}
```

Comment:
I almost changed `lib/mastodon/version.rb` to take care of adding the `+` but opted not to make that change incase it breaks downstream users who currently rely on this specific way of doing the suffixes.

Reply:
Yes, this is why I did not make this change in the initial PR.